### ISSUE TITLE
[49.19] Bug? RFE? Nuclear Missiles fired at ground targets from Aerospace Units

### ISSUE THREAD

**Issue by Nehvar** [MegaMek/megamek#5438]
When you enable the advanced aerospace rule for nuclear weapons you can equip ASF bomb slots with Alamo missiles and AR10's/Whiteshark launchers with Santa Anna missiles.  While equipped with these weapons an Aerospace unit on a ground map can fire them at targets on that ground map, including other Aerospace units.  Rather than use the rules for nuclear weapons, however, the rules for Aerospace engagements are used resulting in these attacks doing capital damage to a single location.

Screen images of targeting information for these weapons.

![01_alamo](https://github.com/MegaMek/megamek/assets/59345488/ada43f3a-e4e8-47a4-b6f1-3febf91aff89)
![03_alamo](https://github.com/MegaMek/megamek/assets/59345488/669438ff-d81c-4260-9b09-7b1f2ab16dac)
![05_santaanna](https://github.com/MegaMek/megamek/assets/59345488/c6130fc6-a0dd-4b4f-88f3-5723537b50a4)

Screen images of firing results.

![02_alamo](https://github.com/MegaMek/megamek/assets/59345488/c98bc083-849d-4c77-a8a9-8daeb56b7eb0)
![04_alamo](https://github.com/MegaMek/megamek/assets/59345488/ad3a6b54-a5bc-4b95-ad27-843b13fc0a79)
![06_santaanna](https://github.com/MegaMek/megamek/assets/59345488/87ca1852-3d7c-494d-860d-1a67683ebfc5)

As you can see they did capital-scale damage to the targets instead of nuclear super-happy fun time.  I am unsure if this is a bug.  If not, it's an RFE.

Saves and logs from test runs after firing.

[Nukes_Alamo_Ground_Post.sav.gz](https://github.com/MegaMek/megamek/files/15180917/Nukes_Alamo_Ground_Post.sav.gz)
[Nukes_Alamo_Ground.log](https://github.com/MegaMek/megamek/files/15180918/Nukes_Alamo_Ground.log)

[Nukes_Alamo_Air_Post.sav.gz](https://github.com/MegaMek/megamek/files/15180919/Nukes_Alamo_Air_Post.sav.gz)
[Nukes_Alamo_Air.log](https://github.com/MegaMek/megamek/files/15180922/Nukes_Alamo_Air.log)

[Nukes_SantaAnna_Ground_Post.sav.gz](https://github.com/MegaMek/megamek/files/15180923/Nukes_SantaAnna_Ground_Post.sav.gz)
[Nukes_SantaAnna_Ground.log](https://github.com/MegaMek/megamek/files/15180924/Nukes_SantaAnna_Ground.log)

**Comment by gsparks3** [comment]
Yeah, I think this one is a "feature not fully implemented" - I believe whoever implemented the aerospace nukes originally set them up for space maps, and in atmosphere they (incorrectly) reuse that functionality. Probably not a high priority fix, but a perfectly fair RFE, since the aim is to have all the tabletop rules implemented eventually.

**Comment by Sleet01** [comment]
Since I'm going to remove Princess's ability to use these weapons, I can at least make them fun for the players...

**Comment by HammerGS** [comment]
> Since I'm going to remove Princess's ability to use these weapons, I can at least make them fun for the players...

Don't you mean give her the option to use them :)

I mean can you say no that that face. 
![image](https://github.com/MegaMek/megamek/assets/13752452/c53df326-fd59-4a62-becc-ab0cd76b1581)

**Comment by Sleet01** [comment]
Sshh, I'm using reverse psychology.... robotology?

**Comment by gsparks3** [comment]
Susan Calvin says it's "robopsychology".

**Comment by Sleet01** [comment]
This will be tougher than I expected: there is existing (probably broken) code for handling Alamo bomb shots as artillery, but only when fired as ground-to-ground shots (I think this is actually improperly-applied CapMissile handling?).  In fact, RAW, Alamo "bombs" should fire like AAA or other missiles against air targets, and like Arrow IV indirect shots against ground and hex targets.
I think this means we need to either implement a separate handler for Alamo shots, or move the Cap Damage / Nuclear Blast decision deeper into the existing attack handling code rather than in the Capital Missile attack handler.
